src/k8s/pkg/component/hook.go

```diff
@@ -26,11 +27,7 @@ func networkValues() (map[string]any, error) {
 	}
 
 	// TODO: the cluster cidr should be configurable through a common interface
-	clusterCIDRStr, err := utils.GetServiceArgument("kube-proxy", "--cluster-cidr")
-	if err != nil {
-		return nil, fmt.Errorf("failed to get cluster cidrs from kube-proxy arguments: %w", err)
-	}
-
+	clusterCIDRStr := snap.GetServiceArgument(snap.Instance(), "kube-proxy", "--cluster-cidr")
```

Comment:
what about possible errors herer?

Reply:
Then an empty string is returned. I copied that behavior over from the cluster-agent. IMHO there is not much that a user can do that requires to distinguish between an empty value and an error